gaiaflex/giving-life-smart-contract.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Giving Life to a Smart Contract over Governance
+---
+
+# Giving Life to a Smart Contract with Voting
+
+## Getting the contract
+
+There are two options to get the sample contract:
+
+1. Download [source code](https://github.com/CosmWasm/cosmwasm-plus/tree/v0.1.1/contracts/cw20-base), and [compile](./../getting-started/compile-contract.md) it your self.
+
+2. Download [pre-compiled binary](https://github.com/CosmWasm/cosmwasm-plus/releases/download/v0.1.1/cw20_base.wasm).
+
+::: warning
+What could go wrong with
+downloading and executing a binary uploaded by some random people on the internet right 😉
+:::
+
+### Uploading Contract and Submitting Proposals
+
+This command will handle the deployment of the contract and submission of a proposal and extract **Code ID** from the
+response for later use:
+
+This is the signature of wasm code store submit proposal:
+
+```shell script
+gaiaflex tx gov submit-proposal wasm-store [wasm file] \
+  --source [source] --builder [builder] --title [text] \
+  --description [text] --run-as [address] [flags]
+```
+
+`--run-as` flag is the contract creator, and it is required.
+
+If you run `gaiaflex tx gov submit-proposal wasm-store -h`, you will notice two more important flags:
+
+```shell
+--instantiate-everybody string      Everybody can instantiate a contract from the code, optional
+--instantiate-only-address string   Only this address can instantiate a contract instance from the code, optional
+```
+
+By default the first flag is enabled. If you want only one address to be able to initiate the contract,
```

Comment:
By default no one can instantiate this proposal except a governance vote.

If either of these flags are set, the voting committee should decide if that is acceptable for the given contract. Instantiate-everybody might make sense for a multisig (everyone makes their own), but not for creating a new token.

^^^^ Good to add some of this text